pkg/bindings/images/images.go

```diff
@@ -305,28 +292,28 @@ func Push(ctx context.Context, source string, destination string, options entiti
 }
 
 // Search is the binding for libpod's v2 endpoints for Search images.
-func Search(ctx context.Context, term string, opts entities.ImageSearchOptions) ([]entities.ImageSearchReport, error) {
+func Search(ctx context.Context, term string, options *SearchOptions) ([]entities.ImageSearchReport, error) {
+	if options == nil {
+		options = new(SearchOptions)
+	}
 	conn, err := bindings.GetClient(ctx)
 	if err != nil {
 		return nil, err
 	}
-	params := url.Values{}
-	params.Set("term", term)
-	params.Set("limit", strconv.Itoa(opts.Limit))
-	params.Set("noTrunc", strconv.FormatBool(opts.NoTrunc))
-	params.Set("listTags", strconv.FormatBool(opts.ListTags))
-	for _, f := range opts.Filters {
-		params.Set("filters", f)
+	params, err := options.ToParams()
+	if err != nil {
+		return nil, err
 	}
+	params.Set("term", term)
 
-	if opts.SkipTLSVerify != types.OptionalBoolUndefined {
-		// Note: we have to verify if skipped is false.
-		verifyTLS := bool(opts.SkipTLSVerify == types.OptionalBoolFalse)
-		params.Set("tlsVerify", strconv.FormatBool(verifyTLS))
+	// Note: we have to verify if skipped is false.
+	if options.SkipTLSVerify != nil {
+		params.Del("SkipTLSVerify")
+		params.Set("tlsVerify", strconv.FormatBool(!options.GetSkipTLSVerify()))
```

Comment:
Looks like you do this tls check in a number of places.  Would it make sense to do a utility function?